### PR TITLE
TBOX-214: Actually use temporary directories in logging tests

### PR DESCRIPTION
**Pull request by skalish**
# ↪️ Pull Request

This PR corrects some of our testing to use temporary files correctly.
Rather than using
`logger = tamr_toolbox.utils.logger.create(..., log_directory=tempfile.gettempdir())`
which creates leftover files, it uses
`with tempfile.TemporaryDirectory() as tempdir:`
which creates a directory that is cleaned up post-testing.

## ✔️ PR Todo

- [ ] ~~Add documentation~~
- [ ] ~~Add examples~~
- [ ] Ensure you are following the practices in the [contributor guide](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit) and [coding_standards](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards)
  * [Use consistent and descriptive name conventions](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.477s36w4rds9) 
  * [Uses google-style docstrings](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Google-style-docstrings)
  * [Uses versioned API/client methods only](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.yv6df99fyrq2)
  * [Avoid global variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-global-variables)
  * [Avoid giant try/catch](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-giant-try-/-except)
  * [Don't use mutable default variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Don%E2%80%99t-use-mutable-default-arguments)
  * [Use a `main()` function in your scripts](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Use-a-%E2%80%9Cmain()%E2%80%9D--function-in-your-scripts)
- [x] Added/updated testing for this change (ensure you are testing any changes/new code!)
- [x] Update title to link to TBOX issue (i.e. start pull request title with TBOX-xx) 
- [ ] Approval from first code-reviewer
- [ ] Approval from second code-reviewer (only required if a second reviewer has left comments)
- [ ] Pass all checks

### CODE REVIEW

**Comment by skalish** [comment]
Marking as a draft while resolving Windows testing issues.

**Comment by skalish** [comment]
Testing issues are now resolved.

File: tests/filesystem/test_bash.py [hunk 1]
```diff
@@ -9,22 +8,23 @@
 from datetime import datetime, timedelta
 from typing import Tuple, List
 
-TEMP_DIR = Path(tempfile.gettempdir())
-
 
-def _make_old_files(directory_path: str, num_days_old: int) -> Tuple[List, List]:
+def _make_old_files(
+    directory_path: str, num_days_old: int, temporary_directory: Path
```
Comment (Kovacsova) [inline tests/filesystem/test_bash.py:13]: what's directory_path?

File: tests/filesystem/test_bash.py [hunk 2]
```diff
@@ -124,132 +130,144 @@ def test_failure_to_create_directory_with_relative_path():
...
+            assert path.exists(
+                directory
+            ), f"Directory {directory} removed due to top-level directory removal"
 
-    bash.remove_directories([top_level_directory])
-
-    for directory in directories:
-        assert not path.exists(directory), f"Removed directory {directory} still exists"
+        # Clean up
+        bash.remove_directories(directories)
```
Comment (Kovacsova) [inline tests/filesystem/test_bash.py:164]: wouldn't these be removed as part of the temporary directory?

File: tests/filesystem/test_bash.py [hunk 3]
```diff
@@ -124,132 +130,144 @@ def test_failure_to_create_directory_with_relative_path():
...
 
-    assert not path.exists(
-        top_level_directory
-    ), f"Directory {top_level_directory} still exists after removal"
+        bash.remove_directories([top_level_directory])
```
Comment (Kovacsova) [inline tests/filesystem/test_bash.py:166]: same as comment above

**Review by Kovacsova — APPROVED** [review]
LGTM

File: tests/filesystem/test_bash.py (hunk 1 above)
Comment (skalish) [reply tests/filesystem/test_bash.py:13]: This is to allow for creation of more specific paths within the temporary directory. I'll add it to the docstring for this helper function.

**Pull request merged**